test/json/json_generator_test.rb

```diff
@@ -443,6 +443,20 @@ def test_invalid_encoding_string
       "\x82\xAC\xEF".to_json
     end
     assert_includes error.message, "source sequence is illegal/malformed utf-8"
+
+    error = assert_raise(JSON::GeneratorError) do
+      JSON.dump("\x82\xAC\xEF")
+    end
+    assert_includes error.message, "source sequence is illegal/malformed utf-8"
+
+    # These pass on the pure-Ruby generator but not with the native extension
+    # assert_raise(Encoding::UndefinedConversionError) do
+    #   "\x82\xAC\xEF".b.to_json
+    # end
+    #
+    # assert_raise(Encoding::UndefinedConversionError) do
+    #   JSON.dump("\x82\xAC\xEF".b)
+    # end
```

Comment:
You can conditionally run it with `if defined?(JSON::Pure)`.

I'll be trying to make implementations converge over time, but short term it's helpful to test the divergent behavior.

Reply:
OK, done.

Reply:
Actually there is already a test for it, so just remove the commented out code I think: https://github.com/ruby/json/blob/f887daa403a7f5ad880c48ef3f6c72615df08a80/test/json/json_generator_test.rb#L448-L456

Reply:
That's different, it's for valid UTF-8 bytes, this is invalid UTF-8 bytes.